docs/RFCS/20220426_rbac_extensions_system_privileges.md

```diff
@@ -0,0 +1,137 @@
+
+We could create descriptors to represent every object we want to grant on. This is not an attractive option at first glance as these descriptors will likely only be used to store privileges and nothing more.
+
+## Syntax for System Level Privileges
+The current proposal is to support `GRANT SYSTEM [privileges...] TO [users...]`.
+We cannot do `GRANT [privileges…] ON SYSTEM TO [users…]` due to the fact that tables can be called system, this would be ambiguous. Using two words, such as SYSTEM ADMIN (can’t think of a good phrase here) would work.
+
+Proposed new system level privileges:
+VIEWACTIVITY (this can be broken down into more privileges)
+VIEWACTIVITYREDACTED
+CANCELQUERY
+MODIFYCLUSTERSETTING
+VIEWCLSUTERSETTING
+VIEWSYSTEMTABLES (can be broken down further if necessary)
+
+
+The syntax is up in the air however, the grant would create or modify the entry in the system.privileges table.
+
+Example: 
+`GRANT SYSTEM MODIFYCLUSTERSETTING TO foo` 
+
+Results in the row
+(foo, 1, “system”, 16384, 0)
+
+The bitmask value for MODIFYCLUSTERSETTING is assumed to be 2 << 13 = 16384 in the above example.
+
+Followed by 
+
+`GRANT SYSTEM VIEWCLUSTERSETTING TO foo` 
+
+Results in the row
+(foo, 1, “system”, 49152, 0)
+
+The bitmask value for VIEWCLUSTERSETTING is assumed to be 2 << 14 = 32768 in the above example. 32768 + 16384 = 49152.
+
+## Support for grants on virtual tables
+One thing worth explicitly calling out is that users have asked for the ability to grant on virtual tables (crdb_internal, pg_catalog). This proposal allows us to support grants on virtual tables and unify the grant experience for tables. Virtual tables have OIDs starting at `MaxUint32 = 1<<32 - 1` counting backwards, an INT (64 bit) column for object id supports this case.
+
+## Adding new privileges
+This proposal allows for CockroachDB engineering teams to add privileges on arbitrarily defined objects as long as those objects can be identified by an object type + object id.
+Once the SQL Experience team completes the work for this framework, it is the responsibility of individual engineering teams to add privileges.
+
+## New Bulkio related privileges
+Right now we have an issue that we do not have granular controls to enable backup / restore and export / import. For example, right now, admin is required to do a full cluster backup. We could add a system level privilege (ie FULLCLUSTERBACKUP) to enable full cluster backups without admin privileges.
+
+For backing up individual database objects right now, we check for CONNECT on databases, SELECT on tables and USAGE on schemas and types. We can potentially augment these in the future? 
+
+## New Jobs related privileges
+The ability to interact with jobs hinges on the coarse grained CONTROLJOB role option. It makes sense to define privileges such as CONTROL on individual job types.
```

Comment:
we are still supporting the coarse grained CONTROLJOB option too right?

Reply:
Likely yes, no immediate plans to deprecate any role options.